xbmc/rendering/dx/DirectXHelper.cpp

```diff
@@ -0,0 +1,186 @@
+    CASEDXGIFORMAT(R8G8B8A8_UNORM)
+    CASEDXGIFORMAT(R8G8B8A8_UNORM_SRGB)
+    CASEDXGIFORMAT(R8G8B8A8_UINT)
+    CASEDXGIFORMAT(R8G8B8A8_SNORM)
+    CASEDXGIFORMAT(R8G8B8A8_SINT)
+    CASEDXGIFORMAT(R16G16_TYPELESS)
+    CASEDXGIFORMAT(R16G16_FLOAT)
+    CASEDXGIFORMAT(R16G16_UNORM)
+    CASEDXGIFORMAT(R16G16_UINT)
+    CASEDXGIFORMAT(R16G16_SNORM)
+    CASEDXGIFORMAT(R16G16_SINT)
+    CASEDXGIFORMAT(R32_TYPELESS)
+    CASEDXGIFORMAT(D32_FLOAT)
+    CASEDXGIFORMAT(R32_FLOAT)
+    CASEDXGIFORMAT(R32_UINT)
+    CASEDXGIFORMAT(R32_SINT)
+    CASEDXGIFORMAT(R24G8_TYPELESS)
+    CASEDXGIFORMAT(D24_UNORM_S8_UINT)
+    CASEDXGIFORMAT(R24_UNORM_X8_TYPELESS)
+    CASEDXGIFORMAT(X24_TYPELESS_G8_UINT)
+    CASEDXGIFORMAT(R8G8_TYPELESS)
+    CASEDXGIFORMAT(R8G8_UNORM)
+    CASEDXGIFORMAT(R8G8_UINT)
+    CASEDXGIFORMAT(R8G8_SNORM)
+    CASEDXGIFORMAT(R8G8_SINT)
+    CASEDXGIFORMAT(R16_TYPELESS)
+    CASEDXGIFORMAT(R16_FLOAT)
+    CASEDXGIFORMAT(D16_UNORM)
+    CASEDXGIFORMAT(R16_UNORM)
+    CASEDXGIFORMAT(R16_UINT)
+    CASEDXGIFORMAT(R16_SNORM)
+    CASEDXGIFORMAT(R16_SINT)
+    CASEDXGIFORMAT(R8_TYPELESS)
+    CASEDXGIFORMAT(R8_UNORM)
+    CASEDXGIFORMAT(R8_UINT)
+    CASEDXGIFORMAT(R8_SNORM)
+    CASEDXGIFORMAT(R8_SINT)
+    CASEDXGIFORMAT(A8_UNORM)
+    CASEDXGIFORMAT(R1_UNORM)
+    CASEDXGIFORMAT(R9G9B9E5_SHAREDEXP)
+    CASEDXGIFORMAT(R8G8_B8G8_UNORM)
+    CASEDXGIFORMAT(BC1_TYPELESS)
+    CASEDXGIFORMAT(BC1_UNORM)
+    CASEDXGIFORMAT(BC1_UNORM_SRGB)
+    CASEDXGIFORMAT(BC2_TYPELESS)
+    CASEDXGIFORMAT(BC2_UNORM)
+    CASEDXGIFORMAT(BC2_UNORM_SRGB)
+    CASEDXGIFORMAT(BC3_TYPELESS)
+    CASEDXGIFORMAT(BC3_UNORM)
+    CASEDXGIFORMAT(BC3_UNORM_SRGB)
+    CASEDXGIFORMAT(BC4_TYPELESS)
+    CASEDXGIFORMAT(BC4_UNORM)
+    CASEDXGIFORMAT(BC4_SNORM)
+    CASEDXGIFORMAT(BC5_TYPELESS)
+    CASEDXGIFORMAT(BC5_UNORM)
+    CASEDXGIFORMAT(BC5_SNORM)
+    CASEDXGIFORMAT(B5G6R5_UNORM)
+    CASEDXGIFORMAT(B5G5R5A1_UNORM)
+    CASEDXGIFORMAT(B8G8R8A8_UNORM)
+    CASEDXGIFORMAT(B8G8R8X8_UNORM)
+    CASEDXGIFORMAT(R10G10B10_XR_BIAS_A2_UNORM)
+    CASEDXGIFORMAT(B8G8R8A8_TYPELESS)
+    CASEDXGIFORMAT(B8G8R8A8_UNORM_SRGB)
+    CASEDXGIFORMAT(B8G8R8X8_TYPELESS)
+    CASEDXGIFORMAT(B8G8R8X8_UNORM_SRGB)
+    CASEDXGIFORMAT(BC6H_TYPELESS)
+    CASEDXGIFORMAT(BC6H_UF16)
+    CASEDXGIFORMAT(BC6H_SF16)
+    CASEDXGIFORMAT(BC7_TYPELESS)
+    CASEDXGIFORMAT(BC7_UNORM)
+    CASEDXGIFORMAT(BC7_UNORM_SRGB)
+    CASEDXGIFORMAT(AYUV)
+    CASEDXGIFORMAT(Y410)
+    CASEDXGIFORMAT(Y416)
+    CASEDXGIFORMAT(NV12)
+    CASEDXGIFORMAT(P010)
+    CASEDXGIFORMAT(P016)
+    CASEDXGIFORMAT(420_OPAQUE)
+    CASEDXGIFORMAT(YUY2)
+    CASEDXGIFORMAT(Y210)
+    CASEDXGIFORMAT(Y216)
+    CASEDXGIFORMAT(NV11)
+    CASEDXGIFORMAT(AI44)
+    CASEDXGIFORMAT(IA44)
+    CASEDXGIFORMAT(P8)
+    CASEDXGIFORMAT(A8P8)
+    CASEDXGIFORMAT(B4G4R4A4_UNORM)
+    CASEDXGIFORMAT(P208)
+    CASEDXGIFORMAT(V208)
+    CASEDXGIFORMAT(V408)
+    CASEDXGIFORMAT(SAMPLER_FEEDBACK_MIN_MIP_OPAQUE)
```

Comment:
UWP uses SDK 10.0.18362.0 which seems not support `SAMPLER_FEEDBACK_MIN_MIP_OPAQUE` and `SAMPLER_FEEDBACK_MIP_REGION_USED_OPAQUE` 

Should be removed or added inside `#ifdef TARGET_WINDOWS_DESKTOP` to able compile UWP

Reply:
Yes I'll remove them. They're for D3D12 features it seems.